keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -373,31 +837,62 @@ The struct in question is IntOrString.
 No
 
 ###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
-The controller-manager will see very negligible and almost un-notoceable increase in cpu usage.
+The controller-manager will see very negligible and almost un-noticeable increase in cpu usage.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
 
 ### Troubleshooting
 
 ###### How does this feature react if the API server and/or etcd is unavailable?
-The RollingUpdate will fail or will not be able to proceed if etcd or apiserver is unavailable and
+The RollingUpdate will fail or will not be able to proceed if etcd or API server is unavailable and
 hence this feature will also be not be able to be used.
 
 ###### What are other known failure modes?
-NA
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+Generalized Max Unavailable Threshold Violations
+
+- The number of unavailable pods during a rolling update exceeds the specified maxUnavailable limit for many Stateful Sets.
+  - Detection: Monitor statefulset_unavailability_violation metric; a significant increase in this metric indicates the failure mode.
```

Comment:
This indicates a general failure mode, not specifically that this feature is failing right?

> spec.replicas - status.readyReplicas > maxUnavailable

could be true due to a failure of the pods unrelated to this feature. That doesn't seem extremely useful in generally debugging this feature.